components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/wsdl/SequenceGenerator.java

```diff
@@ -187,8 +191,23 @@ public static void generateSequencesFromSwagger(String swaggerStr, APIIdentifier
                     String inSequence = template.getMappingInSequence(sequenceMap, operationId, soapAction,
                             namespace, soapNamespace, arraySequenceElements);
                     String outSequence = template.getMappingOutSequence();
-                    saveApiSequences(apiIdentifier, inSequence, outSequence, httpMethod.toString().toLowerCase(),
-                            pathName);
+                    if (isResourceFromWSDL) {
+                        Pattern pattern = Pattern.compile("[{}]");
+                        Matcher hasSpecialCharacters = pattern.matcher(pathName);
+                        if (hasSpecialCharacters.find()) {
+                            String resourcePathName = pathName.split("[{]")[0];
```

Comment:
Is this split correct? "[{]" or should it be "[{"

Reply:
yes, it is added to escape the special character "{"